R/rdeephaven/R/client_wrapper.R

```diff
@@ -1,110 +1,256 @@
-      rbr <- as_record_batch_reader(arrow_tbl)
-      return(private$rbr_to_dh_table(rbr))
-    },
-    tibble_to_dh_table = function(tibbl) {
-      arrow_tbl <- arrow_table(tibbl)
-      return(private$arrow_to_dh_table(arrow_tbl))
-    },
-    df_to_dh_table = function(data_frame) {
-      arrow_tbl <- arrow_table(data_frame)
-      return(private$arrow_to_dh_table(arrow_tbl))
+
+    if (length(extra_headers) != 0) {
+      verify_list("extra_headers", extra_headers, TRUE)
+      for (key in names(extra_headers)) {
+        options$add_extra_headers(key, extra_headers[[key]])
+      }
     }
-  )
+
+    if ((auth_token != "") && (auth_type == "anonymous")) {
+      warning("'auth_token' was set but it will not be used, as 'auth_type' is 'anonymous'.")
+    }
+
+    if ((tls_root_certs != "") && (use_tls == FALSE)) {
+      warning("'tls_root_certs' was set but it will not be used, as 'use_tls is FALSE.")
+    }
+
+    internal_client <- new(INTERNAL_Client,
+      target = target,
+      client_options = options
+    )
+    return(new("Client", .internal_rcpp_object = internal_client))
+  }
+)
+
+### HELPER FUNCTIONS ###
+
+check_for_table <- function(client, name) {
+  return(client@.internal_rcpp_object$check_for_table(name))
+}
+
+### USER-FACING METHODS ###
+
+setGeneric(
+  "open_table",
+  function(client_instance, name) {
+    return(standardGeneric("open_table"))
+  },
+  signature = c("client_instance", "name")
+)
+
+#' @export
+setMethod(
+  "open_table",
+  signature = c(client_instance = "Client", name = "character"),
+  function(client_instance, name) {
+    verify_string("name", name, TRUE)
+    if (!check_for_table(client_instance, name)) {
+      stop(paste0("The table '", name, "' does not exist on the server."))
+    }
+    return(new("TableHandle", .internal_rcpp_object = client_instance@.internal_rcpp_object$open_table(name)))
+  }
+)
+
+setGeneric(
+  "empty_table",
+  function(client_instance, size) {
+    return(standardGeneric("empty_table"))
+  },
+  signature = c("client_instance", "size")
+)
+
+#' @export
+setMethod(
+  "empty_table",
+  signature = c(client_instance = "Client", size = "numeric"),
+  function(client_instance, size) {
+    verify_nonnegative_int("size", size, TRUE)
+    return(new("TableHandle", .internal_rcpp_object = client_instance@.internal_rcpp_object$empty_table(size)))
+  }
+)
+
+setGeneric(
+  "time_table",
+  function(client_instance, period, ...) {
+    return(standardGeneric("time_table"))
+  },
+  signature = c("client_instance", "period")
+)
+
+#' @export
+setMethod(
+  "time_table",
+  signature = c(client_instance = "Client", period = "numeric"),
+  function(client_instance, period, start_time = 0) {
+    verify_any_int("period", period, TRUE)
+    verify_any_int("start_time", start_time, TRUE)
+    return(new("TableHandle", .internal_rcpp_object = client_instance@.internal_rcpp_object$time_table(start_time, period)))
+  }
+)
+
+setGeneric(
+  "as_dh_table",
```

Comment:
I think the initial name of `import_table` was clear after I got used to it. My very first suggestion months ago was `push_table`, which I still think is clearer than `import_table` on what that function is doing. If we're between `to_dh_table` and `as_dh_table`, `as` is more typical of R things, and I would expect the user to read the documentation to know what's going on.

Reply:
`import_table` is being used in Python and Go. I think `import_table` does not really tell you what the function does, and I definitely had to get used to the naming. That said, this may be another case of not wanting to spend time worrying about function names. I would advocate changing all of the API's to `push_to_table` if it wouldn't break anyone, but it probably would.

Reply:
Reading the Go code, it looks like this is a name created in Go and not an underlying gRPC API name.
https://github.com/deephaven/deephaven-core/blob/7ae3a2341522810af7cdc80f9fdcdc43772c80f0/go/pkg/client/flight_stub.go#L86
Same in python:
https://github.com/deephaven/deephaven-core/blob/7ae3a2341522810af7cdc80f9fdcdc43772c80f0/py/client/pydeephaven/session.py#L466
https://github.com/deephaven/deephaven-core/blob/7ae3a2341522810af7cdc80f9fdcdc43772c80f0/py/client/pydeephaven/_arrow_flight_service.py#L18

That gives some more freedom in naming.  Neither `importTable` nor `import_table` appear in our docs.  

Maybe lean to `push_to_table` here and create tickets for the other APIs to consider renaming and deprecating the old names.

Reply:
@jmao-denver should comment.